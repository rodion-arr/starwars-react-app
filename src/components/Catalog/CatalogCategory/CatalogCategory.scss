@use "../../../assets/styles/breakpoints";

.catalog {
  padding: 0 var(--global-padding);
  display: flex;
  flex-direction: column;

  &__title {
    margin: 10px 0;
    padding: 5px 0;
    font-weight: bold;
    text-transform: uppercase;
    border-top: solid var(--main-color) 1px;
    border-bottom: solid var(--main-color) 1px;
  }

  &__title-link {
    text-decoration: none;
    color: inherit;

    &:hover {
      text-decoration: underline;
    }
  }
}

.catalog-items-wrap {
  display: flex;
  flex-direction: column;

  @include breakpoints.for-tablet {
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
  }
}
