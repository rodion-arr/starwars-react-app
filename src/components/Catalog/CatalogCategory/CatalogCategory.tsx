import './CatalogCategory.scss';
import {CatalogItem} from "../CatalogItem/CatalogItem";
import React from "react";
import {Link} from "react-router-dom";

export interface CatalogItemProps {
  image: string;
  title: string;
  url: string;
  id: string;
}

interface Props {
  title: string;
  items: CatalogItemProps[];
  categoryUrl?: string
}

export const CatalogCategory = ({
  title,
  items,
  categoryUrl,
}: Props): JSX.Element | null => {
  if (items.length === 0) {
    return null;
  }

  return (
    <div className="catalog">
      <h2 className="catalog__title">
        {categoryUrl && <Link className="catalog__title-link" to={categoryUrl}>{title}</Link>}
        {!categoryUrl && <>{title}</>}
      </h2>
      <div className="catalog__items catalog-items-wrap">
        {items.map(({
          id,
          image,
          url,
          title,
        }) => <CatalogItem key={id} img={image} title={title} url={url} />)}
      </div>
    </div>
  );
};
