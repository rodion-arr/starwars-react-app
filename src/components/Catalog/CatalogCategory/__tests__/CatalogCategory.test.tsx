import {render} from "@testing-library/react";
import {CatalogCategory} from "../CatalogCategory";
import { BrowserRouter as Router } from "react-router-dom"

describe('CatalogItem', () => {
  it('should render correctly', function () {
     const {container} = render(<Router><CatalogCategory
       title="Unit test"
       items={[{
         title: 'test',
         url: '/test',
         id: '0',
         image: '',
       }]}
       categoryUrl="/unit-test"
     /></Router>)

    expect(container).toMatchSnapshot();
  });

  it('should not render with empty items', function () {
     const {container} = render(<Router><CatalogCategory
       title="Unit test"
       items={[]}
       categoryUrl="/unit-test"
     /></Router>)

    expect(container).toMatchSnapshot();
  });

  it('should not render category link on empty categoryUrl', function () {
     const {container} = render(<Router><CatalogCategory
       title="Unit test"
       items={[{
         title: 'test',
         url: '/test',
         id: '0',
         image: '',
       }]}
     /></Router>)

    expect(container).toMatchSnapshot();
  });
});
