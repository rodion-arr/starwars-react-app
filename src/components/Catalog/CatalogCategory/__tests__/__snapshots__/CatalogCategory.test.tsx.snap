// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogItem should not render category link on empty categoryUrl 1`] = `
<div>
  <div
    class="catalog"
  >
    <h2
      class="catalog__title"
    >
      Unit test
    </h2>
    <div
      class="catalog__items catalog-items-wrap"
    >
      <div
        class="catalog-item"
      >
        <a
          class="catalog-item__link"
          href="/test"
        >
          <div
            class="catalog-item__img-wrap"
          >
            <img
              alt="test"
              class="catalog-item__img"
              loading="lazy"
              src=""
            />
          </div>
          <div
            class="catalog-item__info catalog-item-info"
          >
            <div
              class="catalog-item-info__name"
            >
              test
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`CatalogItem should not render with empty items 1`] = `<div />`;

exports[`CatalogItem should render correctly 1`] = `
<div>
  <div
    class="catalog"
  >
    <h2
      class="catalog__title"
    >
      <a
        class="catalog__title-link"
        href="/unit-test"
      >
        Unit test
      </a>
    </h2>
    <div
      class="catalog__items catalog-items-wrap"
    >
      <div
        class="catalog-item"
      >
        <a
          class="catalog-item__link"
          href="/test"
        >
          <div
            class="catalog-item__img-wrap"
          >
            <img
              alt="test"
              class="catalog-item__img"
              loading="lazy"
              src=""
            />
          </div>
          <div
            class="catalog-item__info catalog-item-info"
          >
            <div
              class="catalog-item-info__name"
            >
              test
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
